packages/ui/src/pages/Register/index.tsx

```diff
@@ -1,63 +1,63 @@
-import React, { FC, FormEventHandler, useState, useCallback } from 'react';
+import React, { FC, FormEventHandler, useState, useCallback, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 import classNames from 'classnames';
+import { LogtoErrorI18nKey } from '@logto/phrases';
 
 import { register } from '@/apis/register';
 import Button from '@/components/Button';
 import Input from '@/components/Input';
 import MessageBox from '@/components/MessageBox';
 import TextLink from '@/components/TextLink';
+import useApi from '@/hooks/use-api';
 
 import styles from './index.module.scss';
 
-export type PageState = 'idle' | 'loading' | 'error';
-
 const App: FC = () => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const [username, setUsername] = useState('');
   const [password, setPassword] = useState('');
-  const [pageState, setPageState] = useState<PageState>('idle');
-  const isLoading = pageState === 'loading';
+
+  const { loading, error, result, fetch: fireRegister } = useApi(register);
 
   const signUp: FormEventHandler = useCallback(
     async (event) => {
       event.preventDefault();
-      setPageState('loading');
-      try {
-        window.location.href = (await register(username, password)).redirectTo;
-      } catch {
-        // TODO: Show specific error after merge into monorepo
-        setPageState('error');
-      }
+      await fireRegister(username, password);
     },
-    [username, password]
+    [username, password, fireRegister]
   );
 
+  useEffect(() => {
+    if (result?.redirectTo) window.location.href = result.redirectTo;
```

Comment:
i think it'll be great to always use curly around `if` clause... i'll add it to our lint rule set

Reply:
strange, the `curly` rule doesn't work here.

Reply:
hmm I did meet some lint remove the curly for single line statement. anyway, let's keep the bracket then.